src/NUnitTestExtractor/NUnitTestExtractorTests/TestExtractorTests.cs

```diff
@@ -0,0 +1,255 @@
+        [TestCaseSource(typeof(ValidDataTestCasesCollection))]
+        [Test]
+        public void WritingToFile_ValidDllAndEveryLevel_GetsWrittenWithProperFormat(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            StringBuilder builder = new StringBuilder();
+
+            InvokeGetTestsUsingFileOutput(builder, dlls, NUnitTestExtractorApp.ParseLevel(level));
+
+            string builderText = builder.ToString();
+            string firstLine = builderText.Substring(0,builderText.IndexOf(Environment.NewLine, StringComparison.OrdinalIgnoreCase) + 1);
+
+            Regex regex = null;
+
+            NUnitTestExtractorApp.Level myLevel = NUnitTestExtractorApp.ParseLevel(level);
+
+            switch (myLevel)
+            {
+                case NUnitTestExtractorApp.Level.Namespace:
+                    regex = new Regex(@"\w+\s\|\s[^\.]");
+                    break;
+
+                case NUnitTestExtractorApp.Level.Class:
+                    regex = new Regex(@"\w+\s\|\s[^\.\.^\.]");
+                    break;
+
+                case NUnitTestExtractorApp.Level.Function:
+                    regex = new Regex(@"\w+\s\|\s[^\.\.^\.^\.]");
+                    break;
+            }
+
+            Assert.That(regex.IsMatch(firstLine));
+        }
+        
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1822:MarkMembersAsStatic")]
+        [TestCaseSource(typeof(ValidDataTestCasesCollection))]
+        [Test]
+        public void WritingToFile_ValidDllAndEveryLevel_FileContainsNoDuplicates(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            StringBuilder builder = new StringBuilder();
+            
+            InvokeGetTestsUsingFileOutput(builder, dlls, NUnitTestExtractorApp.ParseLevel(level));
+
+            string[] lines = builder.ToString().Split('\n');
+           
+            Assert.AreEqual(lines.Length, lines.Distinct().Count());
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Performance", "CA1822:MarkMembersAsStatic")]
+        [TestCaseSource(typeof(ValidDataTestCasesCollection))]
+        [Test]
+        public void WritingToStdout_ValidDllAndEveryLevel_GetsWrittenWithProperFormat(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            using (Process process = SetUpProcess(dll, level))
+            {
+                string line = process.StandardOutput.ReadLine();
+
+                Regex regex = null;
+
+                NUnitTestExtractorApp.Level myLevel = NUnitTestExtractorApp.ParseLevel(level);
+
+                switch (myLevel)
+                {
+                    case NUnitTestExtractorApp.Level.Namespace:
```

Comment:
This switch & Assert looks identical to the one in the other test above.  You could extract it out into a private function to reduce code duplication.  Then the tests would just call something like: `VerifyOutputFormat(myLevel);`